subprojects/logging/src/main/java/org/gradle/api/logging/configuration/LoggingConfiguration.java

```diff
@@ -44,6 +45,20 @@
      */
     void setConsoleOutput(ConsoleOutput consoleOutput);
 
+    /**
+     * Specifies which type of warnings should be written to the console.
+     * @since 4.5
+     */
+    @Incubating
+    WarningType getWarningType();
+
+    /**
+     * Specifies which type of warnings should be written to the console.
```

Comment:
I don't feel like "warningType" is the right word for this. It's not about types of warnings, but the way they are displayed. Maybe "WarningMode"?